csv/src/main/java/com/fasterxml/jackson/dataformat/csv/CsvMapper.java

```diff
@@ -460,6 +487,18 @@ protected void _addSchemaProperties(SerializerProvider ctxt, CsvSchema.Builder b
         BeanDescription beanDesc = ctxt.introspectBeanDescription(pojoType);
         final AnnotationIntrospector intr = ctxt.getAnnotationIntrospector();
         for (BeanPropertyDefinition prop : beanDesc.findProperties()) {
+            if (view != null) {
+                boolean viewVisible = false;
+                for(Class<?> propView : prop.findViews()) {
+                    if (propView.equals(view)) {
```

Comment:
Not sufficient: Views actually check inheritance too (so, say, `CharSequence` as inclusion would be acceptable for View `String`).
You might be able to use databind's `ViewMatcher` for checks:

```
public static ViewMatcher construct(Class<?>[] views) { }
```

and then calling method:

```
        public boolean isVisibleForView(Class<?> activeView) { ...}
```

seems like the proper (and still simple) way to go.

Reply:
Thanks, updated also.